vertx-template-engines/vertx-web-templ-handlebars/src/main/java/io/vertx/ext/web/templ/impl/HandlebarsTemplateEngineImpl.java

```diff
@@ -67,6 +67,12 @@ public HandlebarsTemplateEngine setMaxCacheSize(int maxCacheSize) {
   @Override
   public void render(RoutingContext context, String templateFileName, Handler<AsyncResult<Buffer>> handler) {
     try {
+      // Prepare templateDirectory for partials
+      if (loader.getTemplateDirectory() == null) {
+        loader.setTemplateDirectory(templateFileName.substring(0,
+          templateFileName.length() - Utils.pathOffset(context.normalisedPath(), context).length() + 1 // preserve '/'
```

Comment:
An alternative to this is setting the field in the object directly, but that would require adding an instanceof to TemplateHandlerImpl that passes it for only Handlebars. However, this substring method will work as TemplateHandlerImpl#handle constructs the file path as it currently does.